micrometer-core/src/main/java/io/micrometer/core/aop/TimedAspect.java

```diff
@@ -87,13 +97,33 @@ public Object timedMethod(ProceedingJoinPoint pjp) throws Throwable {
         } finally {
             try {
                 sample.stop(Timer.builder(metricName)
-                        .description(timed.description().isEmpty() ? null : timed.description())
-                        .tags(timed.extraTags())
-                        .tags(EXCEPTION_TAG, exceptionClass)
-                        .tags(tagsBasedOnJoinPoint.apply(pjp))
-                        .publishPercentileHistogram(timed.histogram())
-                        .publishPercentiles(timed.percentiles().length == 0 ? null : timed.percentiles())
-                        .register(registry));
+                                    .description(timed.description().isEmpty() ? null : timed.description())
+                                    .tags(timed.extraTags())
+                                    .tags(EXCEPTION_TAG, exceptionClass)
+                                    .tags(tagsBasedOnJoinPoint.apply(pjp))
+                                    .publishPercentileHistogram(timed.histogram())
+                                    .publishPercentiles(timed.percentiles().length == 0 ? null : timed.percentiles())
+                                    .register(registry));
+            } catch (Exception e) {
+                // ignoring on purpose
+            }
+        }
+    }
+
+    private Object processWithLongTaskTimer(ProceedingJoinPoint pjp, Timed timed, String metricName) throws Throwable {
+        LongTaskTimer longTaskTimer = LongTaskTimer.builder(metricName)
+                .description(timed.description().isEmpty() ? null : timed.description())
+                .tags(timed.extraTags())
+                .tags(tagsBasedOnJoinPoint.apply(pjp))
+                .register(registry);
+
+        LongTaskTimer.Sample sample = longTaskTimer.start();
+
+        try {
+            return pjp.proceed();
+        } finally {
+            try {
+                longTaskTimer.stop(sample.task());
```

Comment:
Can `sample.stop()` be used instead here? Then adding the `task()` method would not be needed.

Reply:
Yes, that's much cleaner, I've changed it.